crates/ott-balancer/src/balancer.rs

```diff
@@ -177,6 +177,78 @@ pub struct BalancerContext {
     pub monoliths: HashMap<MonolithId, BalancerMonolith>,
     pub rooms_to_monoliths: HashMap<RoomName, RoomLocator>,
     pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
+    pub monolith_selection: Box<dyn MonolithSelection + Send + Sync + 'static>,
+}
+// #[derive(Debug, Default)]
+pub struct MonolithRegistry;
```

Comment:
This derive should work fine

Reply:
That's what I thought as well, but the compiler tells me those traits aren't being implemented.

Reply:
Played around with it a bit. The reason this doesn't work is because there is no `Default` for `Box`, and `Default` is derived on `BalancerContext`, so it's trying to derive a `Default` impl for a type it doesn't have. 

The easiest solution is probably to manually impl `Default` for `BalancerContext`

Reply:
That looks like it works. 

Now I have some errors with lifetimes. I'm going to leave it here for today, but I would appreciate quite a bit more help tomorrow since lifetimes are a pretty novel concept to me, and frankly I don't understand them at all.

Thanks for your help today.

Reply:
Rust by example is a pretty good resource for this: https://doc.rust-lang.org/rust-by-example/scope/lifetime.html You might find the "Traits" and "Bounds" examples of particular interest.